app/models/vim_performance_daily.rb

```diff
@@ -106,14 +97,14 @@ def self.process_hourly_for_one_day(recs, options = {})
       end
     end
 
+    return [] if result.empty?
     ts_utc = ts.utc.to_time
 
     # Don't bother rolling up values if day is outside of time profile
     rollup_day = tp.nil? || tp.ts_day_in_profile?(ts)
 
-    results = []
-    result.each_key do |key|
-      int, rtype, rid = key
+    results = result.each_key.collect do |key|
```

Comment:
1. `result`: I agree. will look at this. may better to just get this into here.

```
results.each do |h|
```
will return results at the end.

this is too subtile. I'll add the explicit return at the end